### PR TITLE
Added groupId for iOS keychain sharing

### PR DESCRIPTION
**Pull request by Maleandr**

### CODE REVIEW

**Comment by mogol** [comment]
Hi @Maleandr 
First of all, thanks for the contribution! 

The only concern I have is how platform-specific API should be organised, as there could more different options.
E.g. one option is to rename `groupId` to `iOSGroupId` to highlight the difference.
```
Future<String> read({@required String key, String groupId})
```
vs
```
Future<String> read({@required String key, String iOSGroupId})
```
Another option can be
```
Future<String> read({@required String key, iOSOptions ios, AndroidOptions android})
```

**Comment by Maleandr** [comment]
Separate system options looks nice. I will refactor accordingly.

File: lib/flutter_secure_storage.dart [hunk 1]
```diff
@@ -12,40 +13,78 @@ class FlutterSecureStorage {
...
+  Future<void> deleteAll({iOSOptions iOptions, AndroidOptions aOptions}) =>
+      _channel.invokeMethod('deleteAll', <String, dynamic>{'options': _selectOptions(iOptions, aOptions)});
+
+  /// Select correct options based on current platform
+  Map<String, String> _selectOptions(iOSOptions iOptions, AndroidOptions aOptions) {
+    return Platform.isIOS ? iOptions?.params : aOptions?.params;
+  }
+}
+
+class Options {
```
Comment (mogol) [inline lib/flutter_secure_storage.dart]: Can it be `abstract class Options`? 

```
abstract class Options {
  Map<String, String> get params => _toMap();
  Map<String, String> _toMap();
}
```

Comment (Maleandr) [reply lib/flutter_secure_storage.dart]: Yep

**Comment by mogol** [comment]
Thanks!

**Pull request merged**